fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashActivity.java

```diff
@@ -233,59 +101,83 @@ public static DigitalCashViewModel obtainViewModel(FragmentActivity activity) {
   }
 
   public void setupNavigationBar() {
-    bottomNavigationView = findViewById(R.id.digital_cash_nav_bar);
-    bottomNavigationView.setOnItemSelectedListener(
+    navbar = findViewById(R.id.digital_cash_nav_bar);
+    navbar.setOnItemSelectedListener(
         item -> {
           int id = item.getItemId();
-          if (id == R.id.home_coin) {
-            mViewModel.openHome();
-          } else if (id == R.id.history_coin) {
-            mViewModel.openHistory();
-          } else if (id == R.id.send_coin) {
-            mViewModel.openSend();
-          } else if (id == R.id.receive_coin) {
-            mViewModel.openReceive();
-          } else if (id == R.id.issue_coin) {
-            mViewModel.openIssue();
+          if (id != mViewModel.getCurrentSelectedItem().getValue()) {
+            // This prevents the update to be done multiple times. It is done here rather than
+            // in viewModel because otherwise this would be executed twice
+            mViewModel.setCurrentSelectedItem(id);
+          } else {
+            if (id == R.id.digital_cash_home_menu) {
+              Log.d(TAG, "Opening home fragment");
+              openHomeFragment();
+            } else if (id == R.id.digital_cash_history_menu) {
+              Log.d(TAG, "Opening history fragment");
+              openHistoryFragment();
+            } else if (id == R.id.digital_cash_send_menu) {
+              Log.d(TAG, "Opening send fragment");
+              openSendFragment();
+            } else if (id == R.id.digital_cash_receive_menu) {
+              Log.d(TAG, "Opening receive fragment");
+              openReceiveFragment();
+            } else if (id == R.id.digital_cash_issue_menu) {
+              Log.d(TAG, "Trying to open issue fragment");
+              handleOpenIssue();
+            }
```

Comment:
shouldn't this be a switch ? 😁

Reply:
Sadly not.

In java, switch cases must be constants.

But the ids in the R class are not defined as constant at compile time. So we need to use this ugly if/else

A possible solution would be to use an enum to represent the tabs. But I think we will do that in a later phase of the refactor.